src/main/webapp/app/markdown-editor/markdown-editor.component.ts

```diff
@@ -0,0 +1,255 @@
+     */
+    setupMarkdownEditor(): void {
+        this.aceEditorContainer.setTheme('chrome');
+        this.aceEditorContainer.getEditor().renderer.setShowGutter(false);
+        this.aceEditorContainer.getEditor().renderer.setPadding(10);
+        this.aceEditorContainer.getEditor().renderer.setScrollMargin(8, 8);
+        this.aceEditorContainer.getEditor().setHighlightActiveLine(false);
+        this.aceEditorContainer.getEditor().setShowPrintMargin(false);
+        this.aceEditorContainer.getEditor().clearSelection();
+    }
+
+    /**
+     * @function parse
+     * @desc Check if domainCommands are contained within the text to decide how to parse the text
+     *       1. If no domainCommands are contained parse markdown to HTML and emit the result to the parent component
+     *       2. Otherwise create an array containing all domainCommands identifier passed on from the client,
+     *       3. Create a copy of the markdown text
+     *       4. Create the regEx Expression which searches for the domainCommand identifier
+     *       5. Go through the copy of the markdown text until it is empty and split it as soon as a domainCommand identifier is found into [command]
+     *           5a. One command can contain text over several lines
+     *           5b. All the text between two identifiers is mapped to the first identifier
+     *       6. Reduce the copy by the length of the command
+     *       7. Call the parseLineForDomainCommand for command and save it into content
+     *       8. Emit the content to parent component to assign the values of the array to the right attributes
+     */
+    parse(): void {
+        /** check if domainCommands are passed on from the parent component */
+        if (this.domainCommands == null || this.domainCommands.length === 0) {
+            /** if no domainCommands contained emit the markdown text converted to html to parent component to display */
+                this.previewTextAsHtml = this.artemisMarkdown.htmlForMarkdown(this.markdown);
+                this.html.emit(this.previewTextAsHtml);
+            return;
+        } else {
+            /** create array with domain command identifier */
+            const possibleCommandIdentifier = this.domainCommands.map(command => command.getOpeningIdentifier());
+            /** create empty array which
+             * will contain the splitted textline with the corresponding domainCommandIdentifier which
+             * will be emitted to the parent component */
+            const commandTextsMappedToCommandIdentifiers = [];
+            /** create a copy of the markdown text to loop trough it and find the domainCommandIdentifier */
+            let copy = this.markdown.slice(0);
+
+            /** create array with the identifiers to use for RegEx by deleting the [] of the domainCommandIdentifiers */
+            const tagNames = possibleCommandIdentifier.map(tag => tag.replace('[', '').replace(']', '')).join('|');
```

Comment:
I don't like the term `tagNames`. I would rather call it something like `commandIdentifiersString`
Also this is not an array, it is actually a string, please change the commend to `create string`